src/proxy/processors/push-action/pullRemote.js

```diff
@@ -19,20 +20,33 @@ const exec = async (req, action) => {
       fs.mkdirSync(action.proxyGitPath, '0755', true);
     }
 
-    const cmd = `git clone ${action.url} --bare`;
-
+    const cmd = `git clone ${action.url}`;
     step.log(`Exectuting ${cmd}`);
 
-    const response = spawnSync('git', ['clone', action.url, '--bare', '--progress'], {
-      cwd: action.proxyGitPath,
-      encoding: 'utf-8',
-    });
-
-    const cloneOutput = response?.stderr;
-    step.log(cloneOutput);
+    const authHeader = req.headers?.authorization;
+    const [username, password] = Buffer.from(authHeader.split(' ')[1], 'base64')
+      .toString()
+      .split(':');
+
+    await git
+      .clone({
+        fs,
+        http,
+        url: action.url,
+        onAuth: () => {
+          return {
+            username: username,
+            password: password,
+          };
+        },
```

Comment:
For a really tiny arrow function returning an object you can use this syntax.
Matching names in the object can also be inlined

```suggestion
        onAuth: () => ({
          username,
          password,
        }),
```